core/bin/zksync_api/src/fee_ticker/validator/watcher.rs

```diff
@@ -17,63 +19,80 @@ pub trait TokenWatcher {
 /// Watcher for Uniswap protocol
 /// https://thegraph.com/explorer/subgraph/uniswap/uniswap-v2
 #[derive(Clone)]
-pub struct UniswapTokenWatcher {
+pub struct CoinGeckoTokenWatcher {
     client: reqwest::Client,
-    addr: String,
+    url: String,
+    chain_id: u64,
     cache: Arc<Mutex<HashMap<Address, BigDecimal>>>,
 }
 
-impl UniswapTokenWatcher {
-    pub fn new(addr: String) -> Self {
+impl CoinGeckoTokenWatcher {
+    pub fn new(url: String, chain_id: u64) -> Self {
         Self {
             client: reqwest::Client::new(),
-            addr,
+            url,
+            chain_id,
             cache: Default::default(),
         }
     }
-    async fn get_market_volume(&mut self, address: Address) -> anyhow::Result<BigDecimal> {
-        // Uniswap has graphql API, using full graphql client for one query is overkill for current task
-        let start = Instant::now();
-
-        let query = format!(
-            "{{token(id: \"{:#x}\"){{totalLiquidity, derivedETH}}}}",
-            address
-        );
 
+    async fn get<T>(&self, query: &str) -> Result<T, anyhow::Error>
+    where
+        T: DeserializeOwned,
+    {
         let raw_response = self
             .client
-            .post(&self.addr)
-            .json(&serde_json::json!({
-                "query": query.clone(),
-            }))
+            .get(query)
+            .header("accept", "application/json")
             .timeout(REQUEST_TIMEOUT)
             .send()
             .await
-            .map_err(|err| anyhow::format_err!("Uniswap API request failed: {}", err))?;
+            .map_err(|err| anyhow::format_err!("CoinGecko API request failed: {}", err))?;
 
         let response_status = raw_response.status();
-        let response_text = raw_response.text().await?;
+        let response_text = raw_response.text().await.unwrap();
 
-        let response: GraphqlResponse = serde_json::from_str(&response_text).map_err(|err| {
+        serde_json::from_str(&response_text).map_err(|err| {
             anyhow::format_err!(
-                "Error: {} while decoding response: {} with status: {}",
+                "Error: {} while decoding response of query: {} with status: {}",
                 err,
-                response_text,
+                &query,
                 response_status
             )
-        })?;
+        })
+    }
 
-        metrics::histogram!("ticker.uniswap_watcher.get_market_volume", start.elapsed());
+    async fn find_platform_for_network(&self) -> anyhow::Result<CGTypes::AssetPlatform> {
```

Comment:
what is the difference between platform and network?

Reply:
hm... I guess if you don't know, then I failed to name it well. It refers to the coingecko platform for the current network (well, chain id).
would `find_coingecko_platform_for_current_chain_id` be clearer perhaps?

Reply:
done btw